src/umpire/resource/HostMpi3SharedMemoryResource.hpp

```diff
@@ -0,0 +1,47 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef __Host_Shared_Memory_Resource_HPP
+#define __Host_Shared_Memory_Resource_HPP
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "mpi.h"
+#include "umpire/resource/MemoryResource.hpp"
+#include "umpire/util/Platform.hpp"
+
+namespace umpire {
+namespace resource {
+
+class HostMpi3SharedMemoryResource : public MemoryResource {
+ public:
+  HostMpi3SharedMemoryResource(const std::string& name, int id, MemoryResourceTraits traits);
+
+  ~HostMpi3SharedMemoryResource();
```

Comment:
Shouldn't the destructor be virtual since it is a derived class?

Reply:
You want the **base** class destructor to be virtual so inherited objects are properly destroyed via a pointer to a base class type (for example). Not necessarily the derived class.

Reply:
Oh, right.

Reply:
No worries. We all get confused sometimes.  😄